src/main/scala/abalone/variant/Variant.scala

```diff
@@ -193,72 +192,86 @@ abstract class Variant private[variant] (
+          - line move : no direction to apply (a line move is just 1 marble moving)
+          - push : 1 direction, and dest contains a marble
+          - side move : 2 direction:
+              - HEAD will contain the direction to apply to move from orig to dest
+              - TAIL will contain the direction to apply to each marble for doing effectively the side move
 
         2. play the move
           - line move :
-              - move from orig to dest
+              - move from orig to dest, without considering the direction
           - push :
             - dest contains a marble
-            - we know the direction :
+            - we have the direction :
               - apply the direction from dest until there is an empty square or being off the board
                 - in case of being off the board, increment the score (capture = true)
               - do the line move (orig to dest)
           - side move :
-            - apply the direction to orig and dest : but still have to find the marble in the middle in case of a move of 3 marbles ?
-    
-      in case of a side move, orig=A and dest=1, how do we determine how to move A and B ?
-    
+            - we received both directions :s
+              - 1st direction : we know how to move through all these marbles from orig to dest
+              - 2nd direction : we know what direction we want to apply to all these 2 or 3 marbles
     */
 
-    if (pieces.contains(dest)) { // push
-      // apply that direction from orig, until we find an empty square or get out of the board (considering a max distance of 3 from dest)
-
-      // then move the 2 marbles :
-      // pieces(finalDest) = pieces(dest)
-      // pieces(dest) = pieces(orig)
-      // pieces(orig) = None
-      capture = true
+    if (directions.size == 0) { // line move
+      updatedPieceMap = pieces + (dest -> pieces(orig)) - (orig)
+    } else if (directions.size == 1) { // push
+      if(dest.dir(directions(0)) == None) {
+        // __(_)o\
+        capture = true
+        updatedPieceMap = pieces + (dest -> pieces(orig)) - orig
+      } else if (!pieces.contains(dest.dir(directions(0)).get)) { 
+        // __(_)o.
+        updatedPieceMap = pieces + (dest.dir(directions(0)).get -> pieces(dest)) + (dest -> pieces(orig)) - orig
+      } else if (dest.dir(directions(0)).get.dir(directions(0)) == None) {
+        // ___oo\
+        capture = true
+        updatedPieceMap = pieces + (dest.dir(directions(0)).get -> pieces(dest)) + (dest -> pieces(orig)) - orig
+      } else  {
+        // ___oo.
+        updatedPieceMap = pieces + (dest.dir(directions(0)).get.dir(directions(0)).get -> pieces(dest)) + (dest -> pieces(orig)) - orig
+      }
+    } else { // side move
+        // from orig, move of the second direction
+        // then move from orig of the first direction and repeat until we have reached dest
+        if ( orig.dir(directions(1)).get.dir(directions(0)).get.index != dest.index ) {
```

Comment:
I think the issue of using .get here is that each time you are using directions or POS.dir() you are getting back an Option[Pos] and then essentially having to check that its a valid position and/or handle the case for None. Perhaps this can be done at the top of the function or condition and/or using match so that you can use the value more easily?
e.g. 
```
orig.dir(directions(1)) match {
    case Some(p) => <now you can use p as a pos>
    case None => <dont update piece map as this is an error?>
}
```